scene/main/canvas_item.cpp

```diff
@@ -403,6 +408,8 @@ void CanvasItem::set_as_top_level(bool p_top_level) {
 
 	if (!is_inside_tree()) {
 		top_level = p_top_level;
+		propagate_call(SNAME("_invalidate_global_transform"));
+		global_invalid = true;
```

Comment:
```suggestion
```
`propagate_call` will call `_invalidate_global_transform` for `this` node too so no need to set `global_invalid = true` separately.

https://github.com/godotengine/godot/blob/06f5b09ca493d1de5dd4bec97bdb6b9ad4fb1d29/scene/main/node.cpp#L1959-L1975

Reply:
Right, I forgot to remove this line .-.